runners/flink/1.10/src/test/java/org/apache/beam/runners/flink/FlinkExecutionEnvironmentsTest.java

```diff
@@ -0,0 +1,533 @@
+    assertThat(
+        ((Configuration) Whitebox.getInternalState(bev, "configuration"))
+            .getInteger(RestOptions.PORT),
+        is(1234));
+
+    options.setFlinkMaster("FE80:CD00:0000:0CDE:1257:0000:211E:729C");
+    bev =
+        FlinkExecutionEnvironments.createBatchExecutionEnvironment(
+            options, Collections.emptyList());
+    assertThat(
+        ((Configuration) Whitebox.getInternalState(bev, "configuration"))
+            .getString(RestOptions.ADDRESS),
+        is("fe80:cd00:0:cde:1257:0:211e:729c"));
+    assertThat(
+        ((Configuration) Whitebox.getInternalState(bev, "configuration"))
+            .getInteger(RestOptions.PORT),
+        is(RestOptions.PORT.defaultValue()));
+  }
+
+  @Test
+  public void shouldSupportIPv6Streaming() {
+    FlinkPipelineOptions options = PipelineOptionsFactory.as(FlinkPipelineOptions.class);
+    options.setRunner(FlinkRunner.class);
+
+    options.setFlinkMaster("[FE80:CD00:0000:0CDE:1257:0000:211E:729C]:1234");
+    StreamExecutionEnvironment sev =
+        FlinkExecutionEnvironments.createStreamExecutionEnvironment(
+            options, Collections.emptyList());
+    assertThat(
+        ((Configuration) Whitebox.getInternalState(sev, "configuration"))
+            .getString(RestOptions.ADDRESS),
+        is("fe80:cd00:0:cde:1257:0:211e:729c"));
+    assertThat(
+        ((Configuration) Whitebox.getInternalState(sev, "configuration"))
+            .getInteger(RestOptions.PORT),
+        is(1234));
+
+    options.setFlinkMaster("FE80:CD00:0000:0CDE:1257:0000:211E:729C");
+    sev =
+        FlinkExecutionEnvironments.createStreamExecutionEnvironment(
+            options, Collections.emptyList());
+    assertThat(
+        ((Configuration) Whitebox.getInternalState(sev, "configuration"))
+            .getString(RestOptions.ADDRESS),
+        is("fe80:cd00:0:cde:1257:0:211e:729c"));
+    assertThat(
+        ((Configuration) Whitebox.getInternalState(sev, "configuration"))
+            .getInteger(RestOptions.PORT),
+        is(RestOptions.PORT.defaultValue()));
+  }
+
+  @Test
+  public void shouldRemoveHttpProtocolFromHostBatch() {
+    FlinkPipelineOptions options = PipelineOptionsFactory.as(FlinkPipelineOptions.class);
+    options.setRunner(FlinkRunner.class);
+
+    for (String flinkMaster :
+        new String[] {
+          "http://host:1234", " http://host:1234", "https://host:1234", " https://host:1234"
+        }) {
+      options.setFlinkMaster(flinkMaster);
+      ExecutionEnvironment sev =
+          FlinkExecutionEnvironments.createBatchExecutionEnvironment(
+              options, Collections.emptyList());
+      assertThat(
+          ((Configuration) Whitebox.getInternalState(sev, "configuration"))
+              .getString(RestOptions.ADDRESS),
+          is("host"));
+      assertThat(
+          ((Configuration) Whitebox.getInternalState(sev, "configuration"))
+              .getInteger(RestOptions.PORT),
+          is(1234));
```

Comment:
We could avoid duplication of this file and just have a check method which branches depending on whether we have 1.10 or a version below.